### PR TITLE
Compute client

### LINKED ISSUE #39: Add "status" endpoint under computes/xxx that shows status without restarting
This endpoint will be a POST that takes the same JSON as other endpoints and returns both a job ID (hash) and status like the parent endpoint with two notable differences:
1. Job will not be automatically started
2. Status includes an additional option if no job ever existed with this config: "does-not-exist" or similar

### PR DESCRIPTION
**Pull request by ryanrdoherty**

### CODE REVIEW

File: schema/url/computes/alphaDiv.raml [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-    additionalProperties: false
+  # how do we want to ensure the collectionVariable is a collection variable?
+  # presumably the compute service will check and err
+  # but do we want to talk about constraints?
+  AlphaDivComputeConfig:
+    type: ComputeConfigBase
     properties:
-      outputEntityId:
-        type: string
       collectionVariable: VariableSpec
```
Comment (d-callan) [inline schema/url/computes/alphaDiv.raml:16]: Presumably this should say Collection spec I guess

File: schema/url/computes/betaDiv.raml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
-      outputEntityId:
-        type: string
       collectionVariable: VariableSpec
       betaDivDistanceMethod: BetaDivDistanceMethod
 
   BetaDivDistanceMethod:
     type: string
-    enum: ['bray', 'jaccard', 'jsd']
\ No newline at end of file
+    # RD: what happened to these (were in data service)? 'unifrac', 'wunifrac', 'jensen-shannon'
```
Comment (d-callan) [inline schema/url/computes/betaDiv.raml:18]: I don't think we're ready for those. I have an old memory, which may be incorrect, that they rely on building a phylogenetic tree. That's added complexity and brings up questions we decided weren't critical we had answers to for a first release of beta div.

**Review by d-callan — COMMENTED** [review]
Mostly trivial comments so far

File: src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java [hunk 3]
```diff
@@ -51,13 +51,13 @@ protected void execute() {
...
-                                                    util.singleQuote(computeEntityIdColName) + ", " + 
-                                                    util.singleQuote(method) + ")");
+                                                    PluginUtil.singleQuote(computeEntityIdColName) + ", " +
+                                                    PluginUtil.singleQuote(method) + ")");
       String dataCmd = "readr::format_tsv(alphaDivDT)";
       String metaCmd = "getMetadata(alphaDivDT)";
 
       getWorkspace().writeDataResult(connection.eval(dataCmd).asString());
-      getWorkspace().writeMetaResult(connection.eval(metaCmd).asString());
+      getWorkspace().writeStatisticsResult(connection.eval(metaCmd).asString());
```
Comment (d-callan) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java]: Why was this changed from meta to stats?

File: src/main/java/org/veupathdb/service/eda/compute/plugins/example/ExamplePlugin.java [hunk 4]
```diff
@@ -1,78 +1,100 @@
...
+  private static ComputedVariableMetadata createMetadataObject(String computedColumnName) {
+    ComputedVariableMetadata meta = new ComputedVariableMetadataImpl();
+
+    // skip collections for the example
+    meta.setComputedCollections(Collections.emptyList());
+
+    // give the generated variable a role in the viz plugin
+    PlotReference pr = new PlotReferenceImpl();
+    pr.setComputedVariableId(computedColumnName);
+    pr.setVariablePlotRef("appendedValue");
```
Comment (d-callan) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/example/ExamplePlugin.java]: Idk what this is... What does appendedValue mean?

**Review by d-callan — COMMENTED** [review]
I think I'm understanding how the metadata works between compute and data services. I'm not sure we're on the same page about plot refs though, and the meta for the mbio apps is coming from R right now which you seem to have changed.

**Review by d-callan — APPROVED** [review]

File: src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java (hunk 3 above)
Comment (ryanrdoherty) [reply src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java]: Yeah, this looks like a bug.  Does this work?
